runtime/tests/integration_test.rs

```diff
@@ -0,0 +1,235 @@
+		.unwrap();
+
+		let mut ext = sp_io::TestExternalities::new(t);
+		ext.execute_with(|| System::set_block_number(1));
+		ext
+	}
+}
+
+const ALICE: [u8; 20] = [4u8; 20];
+const BOB: [u8; 20] = [5u8; 20];
+const CHARLIE: [u8; 20] = [6u8; 20];
+const DAVE: [u8; 20] = [7u8; 20];
+
+fn origin_of(account_id: AccountId) -> <Runtime as frame_system::Config>::Origin {
+	<Runtime as frame_system::Config>::Origin::signed(account_id)
+}
+
+fn root() -> <Runtime as frame_system::Config>::Origin {
+	<Runtime as frame_system::Config>::Origin::root()
+}
+
+#[test]
+fn join_collator_candidates() {
+	ExtBuilder::default()
+		.balances(vec![
+			(AccountId::from(ALICE), 2_000 * GLMR),
+			(AccountId::from(BOB), 2_000 * GLMR),
+			(AccountId::from(CHARLIE), 1_100 * GLMR),
+			(AccountId::from(DAVE), 1_000 * GLMR),
+		])
+		.staking(vec![
+			// collators
+			(AccountId::from(ALICE), None, 1_000 * GLMR),
+			(AccountId::from(BOB), None, 1_000 * GLMR),
+			// nominators
+			(
+				AccountId::from(CHARLIE),
+				Some(AccountId::from(ALICE)),
+				50 * GLMR,
+			),
+			(
+				AccountId::from(CHARLIE),
+				Some(AccountId::from(BOB)),
+				50 * GLMR,
+			),
+		])
+		.build()
+		.execute_with(|| {
+			assert_noop!(
+				ParachainStaking::join_candidates(origin_of(AccountId::from(ALICE)), 1_000 * GLMR,),
+				parachain_staking::Error::<Runtime>::CandidateExists
+			);
+			assert_noop!(
+				ParachainStaking::join_candidates(
+					origin_of(AccountId::from(CHARLIE)),
+					1_000 * GLMR
+				),
+				parachain_staking::Error::<Runtime>::NominatorExists
+			);
+			assert!(System::events().is_empty());
+			assert_ok!(ParachainStaking::join_candidates(
+				origin_of(AccountId::from(DAVE)),
+				1_000 * GLMR,
+			));
+			assert_eq!(
+				last_event(),
+				Event::parachain_staking(parachain_staking::Event::JoinedCollatorCandidates(
+					AccountId::from(DAVE),
+					1_000 * GLMR,
+					3_100 * GLMR
+				))
+			);
+		});
+}
+
+#[test]
+fn transfer_to_stake() {
+	ExtBuilder::default()
+		.balances(vec![(AccountId::from(ALICE), 3_000 * GLMR)])
+		.build()
+		.execute_with(|| {
+			// CHARLIE has no balance => fails to stake
+			assert_noop!(
+				ParachainStaking::join_candidates(
+					origin_of(AccountId::from(CHARLIE)),
+					1_000 * GLMR,
+				),
+				DispatchError::Module {
+					index: 3,
+					error: 3,
+					message: Some("InsufficientBalance")
+				}
+			);
+			// Alice stakes to become a collator candidate
+			assert_ok!(ParachainStaking::join_candidates(
+				origin_of(AccountId::from(ALICE)),
+				1_000 * GLMR,
+			));
+			// Alice transfer from free balance 1000 GLMR to Bob
+			assert_ok!(Balances::transfer(
+				origin_of(AccountId::from(ALICE)),
+				AccountId::from(BOB),
+				2_000 * GLMR,
+			));
+			assert_eq!(Balances::free_balance(AccountId::from(BOB)), 2_000 * GLMR,);
+			// Bob transfers 1000 GLMR to Charlie via EVM
+			use sp_core::U256;
+			use std::convert::TryInto;
+			let gas_limit = 1u64;
+			let gas_price: U256 = (5 * GLMR).into();
```

Comment:
```suggestion
			let gas_limit = 9999u64;
			let gas_price: U256 = U256::from_dec_str("1".into());
```

Is it failing because it is out of gas? The gas_limit was 1.